### PR TITLE
Alternate version of module (new protocol), and some edits to README.md

### PR DESCRIPTION
**Pull request by mhvwerts**
Hello Arthur,
I hope you are well. Some time ago, I made some corrections to your README file for serpy.
Here's the "pull request"
Best wishes,
Martin

### CODE REVIEW

**Comment by ArthurLuciani** [comment]
Hi, I hadn't seen your pull request as hardly connect to github anymore and I don't often check the ens rennes mail anymore so I must have missed the notification. 

Well, I'm gonna look into this PR and I will had you as developer or higher to the project.

File: serpy2/serpy2.py [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+            btype = FLOATTYPECODE
+        elif type(data) == int:
+            bdata = data.to_bytes(8, 'little', signed = True)
+            btype = INTTYPECODE
+        elif type(data) == str:
+            bdata = data.encode('utf-8')
+            btype = STRTYPECODE
+        elif type(data) == bytes:
+            bdata = data
+            btype = RAWBYTESTYPECODE
```
Comment (ArthurLuciani) [inline serpy2/serpy2.py]: Probably should use `isinstance` in case data is a subclass

**Comment by ArthurLuciani** [comment]
Otherwise seems good

**Comment by mhvwerts** [comment]
Hi Arthur,
Thanks for your reply and comments. I didn't realize that once a pull request has been opened any subsequent commits to the associated branch are automatically added to the PR, so instead of a few corrections to typos you found a bunch of code as well.
Concerning the new code, I still have some details to check and modify. I'll try to do that quickly so that we can close this PR. Remember that I am not a real programmer and I only work on this occasionally, depending on what we need in the lab.
Anyways, it was really a good idea of you to put this nice code library on GitHub. Now we can use it again for having computers and scientific instrumentation talk to each other. Thank you once again for this handy library.
Best
Martin

File: serpy2/serpy2.py (hunk 1 above)
Comment (mhvwerts) [reply serpy2/serpy2.py]: Thanks for pointing this out (see also: https://switowski.com/blog/type-vs-isinstance).
However, for now, let's work only with the base classes `float`, `int`, `bytes`, `str` and no sub-classes.
I will change the `==` in `is`.

**Comment by mhvwerts** [comment]
Hi Arthur,
The test client-server scripts for `serpy2` are working, and `serpy2` appears to be functioning correctly, although not fully tested in situations where frequent and random interruptions to the connections occur.
We will start to use the code in a real-life lab setting: this will likely provide a good stress test for the program.
Please merge if you think everything is OK.

**Comment by ArthurLuciani** [comment]
Good job !

**Pull request merged**